components/network/samba/samba.p5m

```diff
@@ -19,37 +19,56 @@
 # CDDL HEADER END
 #
 # Copyright (c) 2011, 2015, Oracle and/or its affiliates. All rights reserved.
+# Copyright 2019 Rouven Weiler
 #
 
-<include samba.p5m.common>
+#
+# directories
+dir  path=etc
+dir  path=etc/samba
+dir  path=etc/samba/private mode=500
+dir  path=usr
+dir  path=usr/include
+dir  path=usr/include/samba
+dir  path=var
+dir  path=var/samba
+dir  path=var/samba/cache
+dir  path=var/samba/com
+dir  path=var/samba/lib
+dir  path=var/samba/lock
+dir  path=var/samba/locks
+dir  path=var/samba/log
+dir  path=var/samba/run
+dir  path=var/spool/samba mode=1777
+
+#
+# smbprofiles
+file usr/share/man/man1/profiles.1 path=usr/share/man/man1/smbprofiles.1
 
-# "samba.proto-objects"-file is created using the command:
-#   gmake sample-manifest METADATA_TEMPLATE=
-# as a "sample-manifest"-file (in build/ dir) and is pkgfmt clean
-# All individual actions and attribute settings on pkg objects should
-# be performed by "<transform ...> (pkgmogrify) actions in "samba.p5m.common".
```

Comment:
Should we just merge samba.p5m and samba.proto-objects?

Reply:
I would not. The idea was: One simply runs the gmake sample-manifest as it is without any arguments. This creates the file needed with all its auto-generated header info etc. This file is then included into samba.p5m, which has not to be changed. In samba.p5m some hand-made additions are made. 

Otherwise one always has to do a "special arguments command" (which you might forget to do that way) for a repeatedly needed "gmake sample-manifest". That's why I did it this way.

Reply:
I mean, we can reformat samba.p5m, including samba.proto-objects in it, and treat it as any other component. When you update it, you just look at git diff manifests/sample-manifest.p5m, and update samba.p5m as necessary.

Reply:
Maybe I do not get your idea. (Maybe because it's Monady ;o)

samba.p5m includes samba.proto-objects automatically (see https://github.com/chincheta0815/oi-userland/blob/7e4df109667e92664dc105044d01326d4770c9d7/components/network/samba/samba.p5m#L48). By simply doing a "gmake sample-manifest" plus a "cp ./manifests/sample-manifest.p5m ../samba.proto-objects" the workflow does the same as you want to do, but at least for me, easier, less error prone. (Without the diff and handily checking the files contents, I often forgot to use MANIFEST_TEMPLATE= and caused some issues...)

Maybe I just do not get your thought!?

Reply:
@Mno-hime What do you think? 
I understand that current approach works fine, it's just different from other components, and this worries me.

Reply:
Short addition: 
Yes, the current approach (everything before 4.10.x works fine; a suer tends to do mistakes with it).

The new files were not intended to be changed. After the working on the files they turned out to be more useful. They just grew by modifying the old files into their new layout. A user (me) often does not find the line stating that "MANIFEST_TEMPLATE=" is needed, ending up with searching what caused and error. And useing that argument is quite special to the samba-pkg.

Reply:
I find both ways of building this equally awkward, so I am fine with what the author choose.